modules/analysis-common/src/main/java/org/elasticsearch/analysis/common/CommonAnalysisPlugin.java

```diff
@@ -415,7 +415,11 @@
         filters.add(PreConfiguredTokenFilter.singleton("edge_ngram", false, false, input ->
                 new EdgeNGramTokenFilter(input, 1)));
         filters.add(PreConfiguredTokenFilter.singletonWithVersion("edgeNGram", false, false, (reader, version) -> {
-            if (version.onOrAfter(org.elasticsearch.Version.V_6_4_0)) {
+            if (version.onOrAfter(org.elasticsearch.Version.V_7_0_0)) {
+                throw new IllegalArgumentException(
+                        "The [edgeNGram] token filter name was deprecated in 6.4 and cannot be used in new indices. "
+                                + "Please change the filter name to [edge_ngram] instead.");
+            } else if (version.onOrAfter(org.elasticsearch.Version.V_6_4_0)) {
```

Comment:
Why are we restricting the deprecation to indices created after `6.4.0` ?  Shouldn't we print a deprecation for all indices created before v7 ?

Reply:
That decision was taken already in the original deprecation back in #30209, so I didn't question that again here. I think we didn't want to have people going from 6.3 to 6.4 running into warnings they didn't have before on a minor version change. Does this make sense? I guess we could extend it to all 6.0.0 versions now on a major version break.

Reply:
> I think we didn't want to have people going from 6.3 to 6.4 running into warnings they didn't have before on a minor version change.

I think it's better to log the deprecation for all indices created before v7, otherwise it could be weird for a user to get an error when trying to create an index with the same mapping than v6. So +1 to always log the deprecation warning.

Reply:
Would we want to change that retroactively on existing 6.x branches though?

Reply:
I think it's too late to add a deprecation warning in a patch release. The deprecation log will appear in v7 and should help users to understand why they get an error when they try to copy the mapping in a new index.